src/universe/universe.spec.ts

```diff
@@ -49,3 +51,662 @@ describe('Universe: set hosting environment dynamically', () => {
+
+    const res = await (universe.me as any).getPatchConfig()
+
+    expect((universe as any).me.cachedMeData).toEqual(expectedMe)
+    expect(clientMock).toHaveBeenCalledTimes(1)
+    expect(clientMock).toHaveBeenCalledWith({ method: 'GET', skipInterceptors: false, url: 'https://dev.hello-charles.com/api/v0/me' })
+
+    expect(res.meData).toBe(expectedMe)
+    expect(res.staffRequestOpts).toEqual({
+      method: 'PATCH',
+      url: 'https://dev.hello-charles.com/api/v0/staff/EXPECTED_STAFF_ID',
+      data: []
+    })
+  })
+
+  it('getPatchConfig method relies on cachedMeData when existing and returns expected config', async () => {
+    const expectedMe = {
+      staff: { id: 'EXPECTED_STAFF_ID' }
+    }
+
+    const { universe, clientMock } = genUniverse({
+      clientMockResolveValue: expectedMe,
+      universeCachedMeData: expectedMe
+    })
+
+    expect((universe as any).me.cachedMeData).toBe(expectedMe)
+
+    const res = await (universe.me as any).getPatchConfig()
+
+    expect((universe as any).me.cachedMeData).toEqual(expectedMe)
+    expect(clientMock).toHaveBeenCalledTimes(0)
+
+    expect(res.meData).toBe(expectedMe)
+    expect(res.staffRequestOpts).toEqual({
+      method: 'PATCH',
+      url: 'https://dev.hello-charles.com/api/v0/staff/EXPECTED_STAFF_ID',
+      data: []
+    })
+  })
+
+  it('addTemplateFavorite method can add a template to an existing message_template_favorites and cache it', async () => {
+    const previousFavorite = {
+      id: 'MockId',
+      created_at: 'MockTime',
+      locale: 'fr'
+    }
+    const newFavorite = {
+      id: 'MockId2',
+      created_at: 'MockTime2',
+      locale: 'de'
+    }
+    const previousPreferences = {
+      message_template_favorites: [previousFavorite]
+    }
+    const previousMe = {
+      user: 'UserId',
+      staff: { id: 'EXPECTED_STAFF_ID', preferences: previousPreferences },
+      preferences: previousPreferences
+    }
+
+    const expectedMe = {
+      user: 'UserId',
+      staff: {
+        id: 'EXPECTED_STAFF_ID',
+        preferences: {
+          message_template_favorites: [previousFavorite, newFavorite]
+        }
+      },
+      preferences: {
+        message_template_favorites: [previousFavorite, newFavorite]
+      }
+    }
+
+    const { universe, clientMock } = genUniverse({
+      clientMockResolveValue: [expectedMe.staff],
+      universeCachedMeData: previousMe
+    })
+
+    expect((universe as any).me.cachedMeData).toBe(previousMe)
+
+    const res = await universe.me.addTemplateFavorite(newFavorite)
+
+    expect(clientMock).toHaveBeenCalledTimes(1)
+
+    // eslint-disable-next-line @typescript-eslint/naming-convention, @typescript-eslint/no-unused-vars
+    const { created_at, ...omitCreatedAt } = newFavorite
+    expect(clientMock).toHaveBeenCalledWith({
+      data: [
+        {
+          op: 'add',
+          path: '/preferences/message_template_favorites/-',
+          value: expect.objectContaining(omitCreatedAt)
```

Comment:
I'd suggest we make it explicit, it's not a huge payload. The best tests are usually as explicit, static and obvious as possible. For every occurrence of this `objectContaining`.